google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResultsImpl.java

```diff
@@ -86,9 +94,14 @@ private void sendRequest() {
       // projection entity can represent all type of results
       actualResultType = ResultType.PROJECTION_ENTITY;
     }
+    boolean isQueryPlan = actualResultType.getQueryType() == null && queryMode == QueryMode.PLAN;
     Preconditions.checkState(
-        queryResultType.isAssignableFrom(actualResultType),
+        queryResultType.isAssignableFrom(actualResultType) || isQueryPlan,
         "Unexpected result type " + actualResultType + " vs " + queryResultType);
```

Comment:
Nitpick: Error message can also include scenario when it is not a query plan?

Reply:
fixed, thanks!